### ISSUE TITLE
Lint/UselessAccessModifier - being thrown on class singleton self definition

### ISSUE THREAD

**Issue by katelovescode** [rubocop-hq/rubocop#6076]
When I use the class singleton format, as below, Rubocop tells me it's a useless private access modifier:

```
class MyClass
  def self.class_method
    # stuff
  end

  class << self
    private

    def self.other_class_method
		# stuff
    end
  end
end
```

When I use the `private_class_method` definition format, as below, Rubocop does not tell me it's useless:

```
class MyClass
  def self.class_method
    # stuff
  end

  def self.other_class_method
	# stuff
  end

  private_class_method :other_class_method
end
```

--------

## Expected behavior

Rubocop does not throw a linting error

## Actual behavior

```
W: Lint/UselessAccessModifier: Useless private access modifier.
    private
    ^^^^^^^
```

## Steps to reproduce the problem

Define a private class method using the class singleton method and run rubocop

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -v
0.57.2
```

**Comment by Darhazer** [comment]
It's actually useless, as `other_class_method` is not private at all.
You are confused by the fact that you are still calling `def self.other_class_method`
You have to remove the `self` part. This way you are not defining other_class_method on MyClass, but on it's singleton class, so the way yo call it is `MyClass.singleton_class.other_class_method` (and it's not private!)
